test/unit/test_il_definitions.c

```diff
@@ -0,0 +1,382 @@
+	rz_il_bv_free(res);
+
+	res = rz_il_bv_append_zero(x, 5);
+	mu_assert("append 5 zero", is_euqal_bv(res, append));
+	rz_il_bv_free(res);
+
+	res = rz_il_bv_cut_head(x, 2);
+	mu_assert("cut head 2 zero", is_euqal_bv(res, cut_h));
+	rz_il_bv_free(res);
+
+	res = rz_il_bv_cut_tail(x, 4);
+	mu_assert("cut tail 4 zero", is_euqal_bv(res, cut_t));
+	rz_il_bv_free(res);
+
+	rz_il_bv_free(prep);
+	rz_il_bv_free(append);
+	rz_il_bv_free(cut_h);
+	rz_il_bv_free(cut_t);
+	rz_il_bv_free(x);
+
+	mu_end;
+}
+
+bool test_rzil_bv_cast(void) {
+	ut32 normal, shadow;
+	normal = 2021;
+	shadow = rz_il_bv_to_ut32(rz_il_bv_new_from_ut32(32, normal));
+
+	mu_assert("cast bv<->ut32", normal == shadow);
+	mu_end;
+}
+
+bool test_rzil_bool_init(void) {
+	RzILBool b = rz_il_new_bool(true);
+	mu_assert_notnull(b, "New RzILBool");
+	mu_assert_eq(b->b, true, "bool is true");
+	rz_il_free_bool(b);
+	mu_end;
+}
+
+bool test_rzil_bool_logic(void) {
+	RzILBool t = rz_il_new_bool(true);
+	RzILBool f = rz_il_new_bool(false);
+	RzILBool result;
+
+	// and
+	// t and t => true
+	// f and f => false
+	// t and f => false
+	result = rz_il_bool_and(t, t);
+	mu_assert("true and true", is_equal_bool(result, t));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_and(t, f);
+	mu_assert("true and false", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_and(f, f);
+	mu_assert("false and false", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	// or
+	// t or t => true
+	// t or f => true
+	// f or f => false
+	result = rz_il_bool_or(t, t);
+	mu_assert("true or true", is_equal_bool(result, t));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_or(t, f);
+	mu_assert("true or false", is_equal_bool(result, t));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_or(f, f);
+	mu_assert("false or false", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	// not
+	// not t => false
+	// not f => true
+	result = rz_il_bool_not(t);
+	mu_assert("not true", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_not(f);
+	mu_assert("not false", is_equal_bool(result, t));
+	rz_il_free_bool(result);
+
+	// xor
+	// t xor t => false
+	// f xor f => false
+	// t xor f => true
+	result = rz_il_bool_xor(t, t);
+	mu_assert("t xor t", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_xor(f, f);
+	mu_assert("f xor f", is_equal_bool(result, f));
+	rz_il_free_bool(result);
+
+	result = rz_il_bool_xor(t, f);
+	mu_assert("t xor f", is_equal_bool(result, t));
+	rz_il_free_bool(result);
+
+	rz_il_free_bool(t);
+	rz_il_free_bool(f);
+	mu_end;
+}
+
+bool all_tests() {
+	mu_run_test(test_rzil_bv_init);
+	mu_run_test(test_rzil_bv_cmp);
+	mu_run_test(test_rzil_bv_cast);
+	mu_run_test(test_rzil_bv_operation);
+	mu_run_test(test_rzil_bv_logic);
+	mu_run_test(test_rzil_bv_algorithm);
+
+	mu_run_test(test_rzil_bool_init);
+	mu_run_test(test_rzil_bool_logic);
+	return tests_passed != tests_run;
+}
+
+mu_main(all_tests)
```

Comment:
Thanks, these tests are good! This is exactly what we want because it's very easy to see if the design decisions are good or not. I understand you have many other things on your plate right now, but we should have more of these kind of tests for all the other RzIL API/structures.